### ISSUE TITLE
Distribute dotnet global tool via dotnet and chocolatey

### ISSUE THREAD

**Issue by digitalcoyote** [digitalcoyote/NuGetDefense#10]
I plan to pack this for Chocolatey so it can be used. In the meantime DevAudit should be a good alternative.

**Comment by 616b2f** [comment]
How about using the dotnet tool capability to distribute your tool? It would be also nice because it make it easy to use in in CI pipelines.

**Comment by johnkors** [comment]
Hi! Just stumbled across this. Great work! 

Re CLI: That's what I did with [dotnet-retire](https://github.com/retirenet/dotnet-retire). I actually began the project as a CLI tool, _then_ after a while thought about also making a MSBuild task like done here in NuGet Defense - which absolutely makes sense.

**Comment by johnkors** [comment]
Btw, just so you know as a maintainer. Microsoft is working on a spec for dependency auditing as part of the dotnet SDK. It's been stale for a year or so, but eventually they'll incorporate that into the SDK.

https://github.com/NuGet/Home/wiki/Flag-vulnerable-packages

**Comment by digitalcoyote** [comment]
I'm going to circle back on this after I automate the maintenance (automatically updates the embedded NVD source) deploy process. I intend to make a dotnet tool that can be installed, and write a chocolatey package to allow installation/updating through that package manager on Windows.

**Comment by 616b2f** [comment]
> I'm going to circle back on this after I automate the maintenance (automatically updates the embedded NVD source) deploy process. I intend to make a dotnet tool that can be installed, and write a chocolatey package to allow installation/updating through that package manager on Windows.

Do you plan to provide it also via winget? (Official windows package manager https://github.com/microsoft/winget-cli)

**Comment by digitalcoyote** [comment]
I've not pushed anything to winget, and I don't use it. But I see no reason to exclude it. I'll look into it after I get the chocolatey package together.

**Comment by digitalcoyote** [comment]
I've have a working  dotnet tool (minimal changes to support it without adding to the existing code (a couple lines to give usage details if it's run without arguments). 

 I'm not going to publish it til I get 1.0.9  released and I get it into the ci. 

Chocolatey will be as simple as creating a powershell script that installs it as a dotnet tool. I have a repo that keeps my chocolatey packages up to date, so after writing that, it should update itself.

WinGet will apparently require an installer be built. This also adds complexity to the maintenance process. Because of this, I'll revisit WinGet at a later date.

**Comment by digitalcoyote** [comment]
https://www.nuget.org/packages/NuGetDefense.Tool dotnet tool has been published. I'm going to work on a chocolatey script next.

**Comment by digitalcoyote** [comment]
Closing this since it's available as a dotnet tool and chocolatey has an installer to wrap this for manageability. WinGet's msi requirement will be looked at again later. I've got some higher priority features/issues and some automation for releases that needs to be accomplished before I can look at WinGet. If anyone explicitly needs winget support, let me know in a separate issue and I'll try to work it into my schedule a little sooner.
